docs/architecture/adr-009-proof-size-analysis.md

```diff
@@ -0,0 +1,155 @@
+# ADR 009: New Non-Interactive Default Rules for Reduced Padding
+
+## Changelog
+
+- 14.11.2022: Initial Draft
+
+## Context
+
+ Is it worth it to change non-interactive default rules as a follow-up of [ADR 008](./adr-008-square-size-independent-message-commitments.md)  to decrease the padding? The downside is that the message inclusion proof size will not be as efficient in big square sizes.
+
+ **Note: This analysis implies the implementation of #1004. If the tree over the subtree roots is not a Namespace Merkle Tree then the analysis of both methods has the same proof size.**
+
+- Inter-message padding can be reduced if we can change the non-interactive default rules from this:
+
+    > - Messages that span multiple rows must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
+    > - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+
+    To this: **Messages start at an index that is a multiple of its `msgMinSquareSize`.**
+
+    As an example, we have this diagram. Message 1 is three shares long and is followed by message 2, which is 11 shares long, so the `msgMinSquareSize` of the second message is equal to four. Therefore we have a padding of 5 shares shown in light blue. Furthermore, with the new non-interactive default rule set, a message of size 11 can start in this block at index zero and index three because they are multiples of four. Therefore, we save four shares of padding while retaining the same commitment.
+
+    ![Padding Savings](./assets/padding-savings.png)  
+
+### Defining variables for this Analysis
+
+- n:= Message length in number of shares
+- k:= Square size
+- r:= Number of rows a message fills in a Celestia square.
+
+### The following Questions will be answered
+
+- Why has a message in a bigger square size even with square size independent commitments O(log(n)) proof size?
+- What is the worst-case constructible message in a square to have the biggest impact from switching from O(log(n)) to O(sqrt(n))?
+- Why can we not use the same trick that we used in Question 1 in a single row for more efficient proofs over the row roots?
+- How big is the proof size for this message?
+- What is the worst constructible block with the most amount of padding with old and new non-interactive defaults?
+- Quantify padding and proof size cost.
```

Comment:
[optional] Since the rest of these list items are questions and the heading is question, proposal to rephrase to a question

```suggestion
- What is the quantified padding and proof size cost?
```